tests/Backend/SOX/BranchFileStorageTest.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Test\Backend\SOX;
+
+use Keboola\StorageApi\DevBranches;
+use Keboola\StorageApi\Options\FileUploadOptions;
+use Keboola\Test\StorageApiTestCase;
+use Throwable;
+
+class BranchFileStorageTest extends StorageApiTestCase
+{
+    private DevBranches $branches;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $developerClient = $this->getDeveloperStorageApiClient();
+        $this->branches = new DevBranches($developerClient);
+        foreach ($this->getBranchesForCurrentTestCase($this->branches) as $branch) {
+            try {
+                $this->branches->deleteBranch($branch['id']);
+            } catch (Throwable $e) {
+                // ignore if delete fails
+            }
+        }
+    }
+
+    public function testDeleteBranchDeleteFiles(): void
+    {
+        $newBranch = $this->branches->createBranch($this->generateDescriptionForTestObject());
+        $branchClient = $this->getDeveloperStorageApiClient()->getBranchAwareClient($newBranch['id']);
+        $filePath = __DIR__ . '/../../_data/files.upload.txt';
+
+        $fileId = $branchClient->uploadFile($filePath, (new FileUploadOptions())->setNotify(false)->setFederationToken(true)->setIsPublic(false));
+        $file = $branchClient->getFile($fileId, (new \Keboola\StorageApi\Options\GetFileOptions())->setFederationToken(true));
+        $this->assertEquals(file_get_contents($filePath), file_get_contents($file['url']));
+
+        $s3Client = new \Aws\S3\S3Client([
+            'version' => 'latest',
+            'region' => $file['region'],
+            'credentials' => [
+                'key' => $file['credentials']['AccessKeyId'],
+                'secret' => $file['credentials']['SecretAccessKey'],
+                'token' => $file['credentials']['SessionToken'],
+            ],
+        ]);
+
+        $object = $s3Client->getObject([
+            'Bucket' => $file['s3Path']['bucket'],
+            'Key' => $file['s3Path']['key'],
+        ]);
+        $this->assertEquals(file_get_contents($filePath), $object['Body']);
+
+        $this->branches->deleteBranch($newBranch['id']);
+
+        try {
+            $s3Client->getObject([
+                'Bucket' => $file['s3Path']['bucket'],
+                'Key' => $file['s3Path']['key'],
+            ]);
+            $this->fail('File should not exist');
```

Comment:
@zajca ja som mal este necommitnutu zmenu tak pridavam

Reply:
:see_no_evil: to sem si asi měl všimnůt aj :D